preciceconfigchecker/rules/provide_mesh.py

```diff
@@ -0,0 +1,92 @@
+import networkx as nx
+from networkx import Graph
+from precice_config_graph.nodes import ParticipantNode, MeshNode
+
+from preciceconfigchecker.rule import Rule
+from preciceconfigchecker.severity import Severity
+from preciceconfigchecker.violation import Violation
+
+
+class ProvideMeshRule(Rule):
+    name = "Provided mesh."
+
+    class UnclaimedMeshViolation(Violation):
+        """
+            This class handles a mesh not being provided by any participant.
+        """
+        severity = Severity.ERROR
+
+        def __init__(self, mesh: MeshNode):
+            self.mesh = mesh
+
+        def format_explanation(self) -> str:
+            return f"The mesh {self.mesh.name} does not get provided by any participant."
+
+        def format_possible_solutions(self) -> list[str]:
+            return [f"Please let any participant provide the mesh {self.mesh.name}.",
+                    "Otherwise, please remove it to improve readability."]
+
+    class RepeatedlyClaimedMeshViolation(Violation):
+        """
+            This class handles a mesh being mentioned in a mapping, but multiple participant providing it.
+        """
+
+        severity = Severity.ERROR
+
+        def __init__(self, participants: list[ParticipantNode], mesh: MeshNode):
+            self.mesh = mesh
+            participants_s = sorted(participants, key=lambda participant: participant.name)
+            self.names = participants_s[0].name
+            for i in range(1, len(participants_s) - 1):
+                self.names += ", "
+                self.names += participants_s[i].name
+            # The last participant has to be connected with "and", the others with a comma.
+            self.names += " and "
+            # Name of last participant
+            self.names += participants_s[-1].name
+
+        def format_explanation(self) -> str:
+            return f"The mesh {self.mesh.name} gets provided by participants {self.names}, which is not allowed."
+
+        def format_possible_solutions(self) -> list[str]:
+            return [f"Please let only one participant provide {self.mesh.name}."]
```

Comment:
```suggestion
            return [f"Ensure that only one participant provides {self.mesh.name}."]
```
Similar for other violation